packages/gatsby-theme-docs/src/layouts/release-notes-detail.js

```diff
@@ -54,7 +55,7 @@ const LayoutReleaseNotesDetail = (props) => {
                 </span>
               </Link>
             </SpacingsInline>
-            <ReleaseNote.H1>{props.pageData.title}</ReleaseNote.H1>
+            <H1>{props.pageData.title}</H1>
```

Comment:
I wouldn't name those components `H1`, `H3` etc. Instead we can do something like this:

```js
import ReleaseNoteHeading from '../components/release-note-heading';

<ReleaseNoteHeading size="big" />
<ReleaseNoteHeading size="small" />
```

Then the component can be implemented as:

```js
const Container = styled.div`
  font-weight: ${designSystem.typography.fontWeights.bold};
  line-height: ${designSystem.typography.lineHeights.releaseNoteHeader};
  font-size: ${designSystem.typography.fontSizes.h3};
  color: unset;
  margin: ${props => props.size === 'big' ? designSystem.dimensions.spacings.huge : designSystem.dimensions.spacings.l} 0 0;
`
const ReleaseNoteHeading = (props) => {
  switch (props.size) {
    case 'big':
      return <Container as="h1" {...props} />;
    case 'small':
      return <Container as="h3" {...props} />;
    default:
      return null;
  }
}
```

This way, the parent component does not need to know which "heading" level should be used since the release notes have "special requirements".

Reply:
Using "big" and "small" seems generic, what do you think about "page" and "list"?

Reply:
Passing "page" or "list" as a prop does not make much sense. If you want to stick to those names, I would have 2 components then:

```js
import { ReleaseNotePageTitle, ReleaseNoteListTitle } from '../components/release-note-heading';

<ReleaseNotePageTitle />
<ReleaseNoteListTitle />
```

and

```js
const Container = styled.div`
  font-weight: ${designSystem.typography.fontWeights.bold};
  line-height: ${designSystem.typography.lineHeights.releaseNoteHeader};
  font-size: ${designSystem.typography.fontSizes.h3};
  color: unset;
  margin: ${props => props.as === 'h1' ? designSystem.dimensions.spacings.huge : designSystem.dimensions.spacings.l} 0 0;
`
const ReleaseNotePageTitle = () => <Container as="h1" />;
const ReleaseNoteListTitle = ( => <Container as="h3" />;
```